src-gui/src/renderer/components/pages/swap/swap/init/redeemAddress/RedeemAddress.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button, Typography, Box, Paper } from "@mui/material";
+import ReceiveAddressSelector from "./ReceiveAddressSelector";
+import { isXmrAddressValid } from "utils/conversionUtils";
+import { isTestnet } from "store/config";
+import { buyXmr } from "renderer/rpc";
+import PromiseInvokeButton from "renderer/components/PromiseInvokeButton";
+import { useState } from "react";
+
+export default function RedeemAddress() {
+  const [redeemAddress, setRedeemAddress] = useState<string>("");
+
+  const isValidAddress =
+    redeemAddress && isXmrAddressValid(redeemAddress, isTestnet());
+
+  async function init() {
+    if (!isValidAddress) {
+      return;
+    }
+    await buyXmr(null, redeemAddress);
```

Comment:
The null parameter here should be replaced by the old refund address logic from InitPage